### DISCUSSION TITLE
Duration / Speech Impl

### DISCUSSION THREAD

**Discussion by bkodes** [lucidrains/soundstorm-pytorch#4] [Ideas]
Reference: [PR Comment](https://github.com/lucidrains/soundstorm-pytorch/pull/2#issuecomment-1554693235)

Ideas / Approach to incorporating duration / speech priors
--> e.g [NaturalSpeech2 Impl](https://github.com/lucidrains/naturalspeech2-pytorch/blob/main/naturalspeech2_pytorch/naturalspeech2_pytorch.py#L271)

**Comment by bkodes** [comment]
@lucidrains just creating this discussion to jot ideas as we have them

**Comment by seastar105** [comment]
[InstructTTS](https://arxiv.org/abs/2301.13662) used variance adaptor from fastspeech2 as pitch/duration predictor with their own style encoder. this model is also nar model with neural audio codec. it generates discrete tokens with discrete diffusion. 

will just using standard Text Encoder + pitch/duration predictor with prompt could work nicely? it seems much doable than training good text-to-semantic transformer.
